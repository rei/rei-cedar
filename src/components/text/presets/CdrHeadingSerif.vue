<script setup lang="ts">
import { useCssModule, computed, type PropType } from 'vue';
import CdrText from '../CdrText.vue';

defineOptions({
  name: 'CdrHeadingSerif',
});

type scaleValue = '1'|'2'|'3'|'4'|'5';

const props = defineProps({
  /** 
   * Sets the type scale
   * @type scaleValue
   * @values '1', '2', '3', '4', '5'
   */ 
  scale: { type: String as PropType<scaleValue>, default: '5' },
  /** Toggles the strong variant */
  strong: { type: Boolean, default: false }
});

const typeProperties = computed(() => {
  return {
    '--cdr-heading-serif-font-size': `var(--cdr-type-scale-${props.scale})`,
    '--cdr-heading-serif-line-height': 
      `calc(var(--cdr-type-scale-${props.scale}) * var(--cdr-heading-serif-line-height-ratio))`,
    '--cdr-heading-serif-font-weight': props.strong ? '600' : '400',

  }
});

const baseClass = 'cdr-heading-serif';
const style = useCssModule();
</script>

<template>
  <CdrText
    :class="style[baseClass]"
    :style="typeProperties"
  >
    <slot />
  </CdrText>


</template>

<style module lang="scss">
@import "@rei/cdr-tokens/dist/rei-dot-com/scss/cdr-tokens.scss";
@import "../../../styles/fluid.css";

.cdr-heading-serif {
  @include cdr-text-heading-serif-1200;
  font-size: var(--cdr-heading-serif-font-size);
  line-height: var(--cdr-heading-serif-line-height);
  font-weight: var(--cdr-heading-serif-font-weight);
}
</style>  