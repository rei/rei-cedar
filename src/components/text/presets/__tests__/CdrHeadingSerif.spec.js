import { mount } from '../../../../../test/vue-jest-style-workaround.js';
import CdrHeadingSerif from '../CdrHeadingSerif.vue';


describe('CdrHeadingSerif', () => {
  describe('default configuration', () => {
    let wrapper;
    beforeEach(() => {
      wrapper = mount(CdrHeadingSerif, {
        slots: {
          default: 'paragraph'
        }
      });
    });
    it('renders correctly', () => {
      expect(wrapper.element).toMatchSnapshot();
    });
  });

  describe('passing a scale', () => {
    let wrapper;
    beforeEach(() => {
      wrapper = mount(CdrHeadingSerif, {
        props: { scale: '1' },
        slots: {
          default: 'paragraph'
        }
      });
    });
    it('renders correctly', () => {
      expect(wrapper.element).toMatchSnapshot();
    });
  });

  describe('passing a scale', () => {
    let wrapper;
    beforeEach(() => {
      wrapper = mount(CdrHeadingSerif, {
        props: { strong: true },
        slots: {
          default: 'paragraph'
        }
      });
    });
    it('renders correctly', () => {
      expect(wrapper.element).toMatchSnapshot();
    });
  });

});
