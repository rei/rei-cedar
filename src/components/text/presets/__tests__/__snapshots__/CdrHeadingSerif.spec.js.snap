// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CdrHeadingSerif > default configuration > renders correctly 1`] = `
<h1
  class="cdr-heading-serif"
  style="--cdr-heading-serif-font-size: var(--cdr-type-scale-5); --cdr-heading-serif-line-height: calc(var(--cdr-type-scale-5) * var(--cdr-heading-serif-line-height-ratio)); --cdr-heading-serif-font-weight: 400;"
>
  
  paragraph
  
</h1>
`;

exports[`CdrHeadingSerif > passing a scale > renders correctly 1`] = `
<h1
  class="cdr-heading-serif"
  style="--cdr-heading-serif-font-size: var(--cdr-type-scale-1); --cdr-heading-serif-line-height: calc(var(--cdr-type-scale-1) * var(--cdr-heading-serif-line-height-ratio)); --cdr-heading-serif-font-weight: 400;"
>
  
  paragraph
  
</h1>
`;

exports[`CdrHeadingSerif > passing a scale > renders correctly 2`] = `
<h1
  class="cdr-heading-serif"
  style="--cdr-heading-serif-font-size: var(--cdr-type-scale-5); --cdr-heading-serif-line-height: calc(var(--cdr-type-scale-5) * var(--cdr-heading-serif-line-height-ratio)); --cdr-heading-serif-font-weight: 600;"
>
  
  paragraph
  
</h1>
`;
